IPython/terminal/prompts.py

```diff
@@ -77,3 +83,52 @@ def write_output_prompt(self):
                 self.shell.pt_cli.print_tokens(tokens)
             else:
                 sys.stdout.write(prompt_txt)
+
+    def write_format_data(self, format_dict, md_dict=None):
+        """Write the format data dict to the frontend.
+
+        This version adds support for image/png if the 
+        IPYTHON_IMAGE_VIEWER environment avariable is set.
+
+        Parameters
+        ----------
+        format_dict : dict
+            The format dict for the object passed to `sys.displayhook`.
+        md_dict : dict (optional)
+            The metadata dict to be associated with the display data.
+        """
+        image_data = format_dict.get("image/png")
```

Comment:
> I wanted to avoid a situation where we also have to specify a list of valid MIME types for each viewer. The ones I checked all support png.

That's fair, and I don't think many backend do return anything other than PNG.